text/0000-macro-hygiene-optout.md

```diff
@@ -0,0 +1,221 @@
+- Feature Name: macro_hygiene_optout
+- Start Date: 2018-07-05
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This feature introduces the ability to "opt-out" of the usual macro hygiene rules within definitions of [declarative macros][decl-macro], for designated identifiers or occurrences of identifiers. In other words, the feature will enable one to annotate occurrences of identifiers with macro call-site hygiene rather than the default definition-site hygiene.
+
+# Motivation
+[motivation]: #motivation
+
+The use of [hygienic macros] in Rust is justified by much prior research and experience, and solves several common issues that programmers would otherwise encounter with macros due to the nature of syntactical substitution. The principal deficit of this approach is that it requires that names/identifiers of any items generated by a macro be *explicitly passed to* the macro as arguments. This both requires the logic for name selection to remain entirely external to the macro, and even if that is not a problem, the passing of all identifiers-to-export into a macro can quickly become unwieldy for macros that generate many identifiers.
```

Comment:
> justified by much prior research and experience

A link would be good for curious readers :)

Reply:
I think the "hygienic macros" links offers good justification, no?

Reply:
Truthfully I expected more papers and citations given "much prior research", but I suppose it's enough :)

Reply:
Hah, okay, I'll add one or two!